Consul.Test/LockTest.cs

```diff
@@ -300,7 +300,10 @@ public async Task Lock_ContendFast()
                 }));
             }
 
-            await WithTimeout(Task.WhenAll(tasks));
+            // Set timeout to allow each task to wait up to the lock wait time
+            await WithTimeout(
+                Task.WhenAll(tasks),
+                TimeSpan.FromTicks(2 * contenderPool * Lock.DefaultLockWaitTime.Ticks));
```

Comment:
Now that we are using absolute timeouts everywhere else, why do we use a computed one here?
I think we should be consistent and either compute them everywhere, or nowhere. What do you think?

Reply:
Yeah good point, I think in the cases where the default timeout is overridden it probably makes sense to compute it but with some allowance for extra time rather than just setting it to the maximum expected time as it was before, so I should probably also compute the timeout for `Lock_Contend_LockDelay`.

Although because the timeout is really just there to catch a hang rather than as any kind of performance check, I think it might just make sense to bump up the default timeout to something larger like 5 or 10 minutes and use that everywhere. Would that seem reasonable to you?

Reply:
> Although because the timeout is really just there to catch a hang rather than as any kind of performance check, I think it might just make sense to bump up the default timeout to something larger like 5 or 10 minutes and use that everywhere. Would that seem reasonable to you?

That does sound reasonable to me. Let's go for a default of 5 minutes and add a comment that this is only there to catch a hang, and not a performance check. If we do not need those utility methods to ever be called with a timeout different from the default, then let's remove the optional parameter.